js/router/SharedNavigator.js

```diff
@@ -77,6 +104,33 @@ const SharedNavigator = React.createClass ({
     });
   },
 
+  _onNotification(notification) {
+    const data = loadCachedFormDataById(notification.data.formId);
+    const path = {path: 'form', title: data.survey.title, survey: data.survey, form: data.form}
+    // TODO determine the type of notification
+    // TODO sync remote and cached notifications
+    // addTimeTriggerNotification(data.survey.id, data.form.id, data.form.title, notification.message, new Date());
+    // We will only route the user if notification was remote
+    if (!notification.forground) {
```

Comment:
Small typo here makes this statement always true.
Users might get taken away from their current view even with app on foreground